lib/chef/provider/package/zypper.rb

```diff
@@ -70,7 +71,20 @@ def get_current_versions
         end
 
         def candidate_version
-          @candidate_version ||= package_name_array.each_with_index.map { |pkg, i| available_version(i) }
+          if source_files_exist?
```

Comment:
this helper should really go away and just be `if new_resource.source` checks since if that value is not-nil then you're doing source installs.